### DISCUSSION TITLE
Ansys mesh cannot be used in moose

### DISCUSSION THREAD

**Discussion by liangyu-psi** [idaholab/moose#17434] [Q&A Meshing] [answered]
## Bug Description
The mesh generated by ICEM or Ansys Mesh cannot be recongized by MOOSE.

## Steps to Reproduce
The format of mesh is .msh. It's will cause a dimension error.
But when I use Gmsh. Although the format is still .msh, it will be recongized by MOOSE.

## Impact
It's not convenient to create a mesh by familiar software. I don't know if the mesh file of Ansys does not meet the MOOSE standard. Hope someone can answer. Many thanks.

**Comment by liangyu-psi** [comment]
Is there anybody know the reason?

**Reply by AnthonyB08** [reply]
Is there any update on using Ansys Meshing file for MOOSE?

**Comment by dschwen** [comment, marked as answer]
You might be confusing a file extension with a mesh format. While both files end in .msh they might not be the same format at all. Is there a different format you could export your mesh to with Ansys?

**Reply by cticenhour** [reply]
@dschwen [This link](https://forum.ansys.com/discussion/6362/exporting-mesh-data-from-ansys-workbench) suggests there are only a handful of export options for ANSYS (.mechdat, .msh, .poly, .cgns, .prj, and .tgf), so I think exporting to ANSYS .msh and converting using meshio (in my other answer) is probably the best bet if ANSYS must be used.

**Comment by cticenhour** [comment]
There is a python tool, Meshio, that would allow a conversion from the .msh file created by ANSYS to an Exodus .e file which MOOSE can also import. Check out https://pypi.org/project/meshio/

**Reply by AnthonyB08** [reply]
I have tried this, but i keeps returning the same error of : 

![image](https://user-images.githubusercontent.com/55299169/120042127-811d7f80-bfd7-11eb-97e5-6e492297e398.png)

So then I resorted to importing my .msh into CUBIT and exporting it as a .e

**Reply by cticenhour** [reply]
Oops - Not a very helpful error is it? Though it appears as though meshio could be treating it as a GMSH `.msh` file as well, judging from that output. I don't have ANSYS to troubleshoot with, but I'm glad you have a good workaround!
